Update the Swift version to 6.0!

shahmishal opened the pull request:
The Swift 6.0 release process will be announced soon. The `main` branch will be used to track Swift 6.0 until `release/6.0` branch has been created.

shahmishal committed:
Update the Swift version to 6.0 from 5.11

shahmishal commented:
@swift-ci test

hborla commented on include/swift/AST/ASTContext.h:
```diff
@@ -990,10 +990,10 @@ class ASTContext final {
-  /// Get the runtime availability of features introduced in the Swift 5.11
+  /// Get the runtime availability of features introduced in the Swift 6.0
   /// compiler for the target platform.
   inline AvailabilityContext getSwift511Availability() const {
```
Nit: We should rename `getSwift511Availability()` to `getSwift60Availability()`. That's cosmetic though and can be done in a follow-up PR.

shahmishal replied:
Updated, thanks!

Azoy reviewed (approved):
The standard library changes look good to me! cc: @lorentey for his input.

shahmishal commented:
@swift-ci test

shahmishal commented:
@swift-ci build toolchain

shahmishal commented:
@swift-ci clean test macOS

etcwilde commented:
CC @compnerd, do we need to update the `swift5` section names in https://github.com/apple/swift/blob/main/stdlib/public/runtime/SwiftRT-ELF-WASM.cpp and https://github.com/apple/swift/blob/main/stdlib/public/runtime/SwiftRT-COFF.cpp?

MaxDesiatov commented:
cc @kateinoigakukun for Wasm changes

shahmishal commented:
@swift-ci test Linux

etcwilde committed:
Set version info with `#define`

Don't use `#cmakedefine` to define values that can be zero.
`#cmakedefine` only sets the definition when the corresponding value in
CMake itself has a truthy value. `0` has a false-y value, so
SWIFT_VERSION_MINOR is undefined for 6.0 resulting in some things
breaking.

etcwilde commented:
@swift-ci please test

shahmishal committed:
Set the default compiler effective language version to Swift 5 for now

shahmishal commented:
@swift-ci test

kateinoigakukun commented:
We can rename `swift5` section names for ABI unstable platforms, but it's not mandatory

etcwilde commented on include/swift/Basic/LangOptions.h:
```diff
@@ -172,7 +172,7 @@ namespace swift {
     ///
 
     /// User-overridable language version to compile for.
-    version::Version EffectiveLanguageVersion = version::Version::getCurrentLanguageVersion();
```
This change seems to be causing issues in CI. Out of curiosity, why don't we want the current language version here?
```
/home/build-user/swift/include/swift/Basic/LangOptions.h:175:49: error: use of undeclared identifier 'Version'
    version::Version EffectiveLanguageVersion = Version{5};
                                                ^
/home/build-user/swift/include/swift/Basic/LangOptions.h:175:56: error: expected ';' at end of declaration list
    version::Version EffectiveLanguageVersion = Version{5};
                                                       ^
```

shahmishal replied:
Not all of the swift code is Swift 6 ready yet, I have a patch to fix this issue. Also, working on getting a new PR for `swift-experimental-string-processing`

Previous error:

```
/home/build-user/swift/stdlib/public/libexec/swift-backtrace/main.swift:84:14: error: static property 'args' is not concurrency-safe because it is non-isolated global shared mutable state
  static var args = Arguments()
```

shahmishal commented:
https://github.com/apple/swift-experimental-string-processing/pull/722

@swift-ci test

shahmishal commented:
https://github.com/apple/swift-experimental-string-processing/pull/722

@swift-ci build toolchain

compnerd commented:
> > CC @compnerd, do we need to update the `swift5` section names in https://github.com/apple/swift/blob/main/stdlib/public/runtime/SwiftRT-ELF-WASM.cpp and https://github.com/apple/swift/blob/main/stdlib/public/runtime/SwiftRT-COFF.cpp?
> 
> We can rename `swift5` section names for ABI unstable platforms, but it's not mandatory

I don't think that we should. It would be an unnecessary break. If there was a motivation for doing this, e.g. we are changing the contents of the linker tables, then perhaps. Otherwise, I think that bumping the version is really unnecessary. Particularly on non-Darwin, the collection itself is versioned so that we can continue to add tables without changing the contents. So, even in the case where we want to change the contents, I think that we have other means.

mickael-swiftui commented on cmake/SwiftVersion.cmake:
```diff
@@ -1,7 +1,7 @@
-set(SWIFT_VERSION_MAJOR 5)
-set(SWIFT_VERSION_MINOR 11)
+set(SWIFT_VERSION_MAJOR 6)
+set(SWIFT_VERSION_MINOR 0)
```
I guess it’s the digit corresponding to the 0 of 6.0

etcwilde replied:
That is correct.

shahmishal committed:
Fix the Serialization/version-mismatches.swift Swift version

shahmishal committed:
Merge remote-tracking branch 'origin/main' into bump-swift-version-to-6

lorentey reviewed (approved):
Looks good! 🚢

shahmishal committed:
Merge remote-tracking branch 'origin/main' into bump-swift-version-to-6

shahmishal commented:
https://github.com/apple/swift-experimental-string-processing/pull/722

@swift-ci test

shahmishal committed:
Set the minor version to 10 (5.10) for language version

shahmishal committed:
Merge branch 'main' into bump-swift-version-to-6

shahmishal committed:
Update the SwiftStdlib version to 6.0 and fix Frontend/crash.swift test

shahmishal committed:
Fix the test/Serialization/Recovery/types-5-to-4.swift test string to match Swift version 5.10

shahmishal commented:
https://github.com/apple/swift-experimental-string-processing/pull/722

@swift-ci test

shahmishal committed:
Merge remote-tracking branch 'origin/main' into bump-swift-version-to-6

finagolfin commented on include/swift/Basic/LangOptions.h:
```diff
@@ -171,7 +171,7 @@ namespace swift {
 
     /// User-overridable language version to compile for.
-    version::Version EffectiveLanguageVersion = version::Version::getCurrentLanguageVersion();
+    version::Version EffectiveLanguageVersion = version::Version{5, 10};
```
@shahmishal, do you plan to bump this to 6 soon? Otherwise, I'm seeing strange behavior like this when compiling code with the Swift 6 snapshots:
```
> cat check.swift
#if swift(>=6.0)
print("got to 6.0")
#else
print("no go to 6.0")
#endif
> ./swift-DEVELOPMENT-SNAPSHOT-2024-07-29-a-ubi9/usr/bin/swiftc check.swift  && ./check
no go to 6.0
> ./swift-6.0-DEVELOPMENT-SNAPSHOT-2024-08-02-a-ubi9/usr/bin/swiftc check.swift  && ./check
no go to 6.0
```
Of course, this can be worked around by explicitly setting the language version:
```
> ./swift-DEVELOPMENT-SNAPSHOT-2024-07-29-a-ubi9/usr/bin/swiftc check.swift -swift-version 6 && ./check
got to 6.0
```

shahmishal replied:
This is expected, the Swift 6.0 language mode is not enabled by default. Developers are required to enable Swift 6 language mode by using `-swift-version 6.0` 

https://www.swift.org/migration
> Important
> The Swift 6 language mode is opt-in. Existing projects will not switch to this mode without configuration changes.

finagolfin replied:
OK, I now see [this was discussed on the forum months ago](https://forums.swift.org/t/default-swift-language-version-for-a-given-toolchain-version/71100), which I missed: I will ask about it there.
